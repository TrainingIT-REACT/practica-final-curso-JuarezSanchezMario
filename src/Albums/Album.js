import React, { Component } from "react";
import { connect } from "react-redux";
import { getSongs } from "../reducers/songs";
import { fetchByAlbum } from "../actions/songs";
import { getSelectedAlbum } from "../reducers/albums";
import Card from "react-bootstrap/Card";
import Container from "react-bootstrap/Container";
import ListGroup from "react-bootstrap/ListGroup";
import ListGroupItem from "react-bootstrap/ListGroupItem";
import AudioPlayer from "react-h5-audio-player";
import "react-h5-audio-player/lib/styles.css";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

// Css
import "./Albums.css";

class Album extends Component {
  constructor(props) {
    super(props);
    this.duration = 0;
    this.state = {
      selectedSong: {
        source: {},
        id: {},
        last: false
      }
    };
  }

  componentDidMount() {
    this.props.fetchSongs(this.props.selectedAlbum.id);
  }

  render() {
    const ponerCancion = (source, id) => {
      this.setState(
        {
          selectedSong: {
            source: ""
          }
        },
        () =>
          this.setState({
            selectedSong: {
              source: source,
              id: id,
              last:
                id === this.props.songs[this.props.songs.length - 1].id
                  ? true
                  : false
            }
          })
      );
    };

    if (!this.props.songs.length) {
      return "nope";
    }
    return (
      <div className="Albums">
        <Container style={{ "text-align": "-webkit-center" }}>
          <Row>
            <Col>
              <Card style={{ width: "18rem" }}>
                <Card.Img variant="top" src={this.props.selectedAlbum.cover} />
                <Card.Body>
                  <Card.Title>{this.props.selectedAlbum.name}</Card.Title>
                  <Card.Text>{this.props.selectedAlbum.artist}</Card.Text>
                  <Card.Text>
                    {this.duration === 0
                      ? this.props.songs.forEach(
                          item => (this.duration += item.seconds)
                        )
                      : null}
                    {new Date(this.duration * 1000).toISOString().substr(14, 5)}
                    {" m"}
                  </Card.Text>
                </Card.Body>
              </Card>
              <Row>
                <AudioPlayer
                className="player"
                  autoPlay={this.state.selectedSong.source.length}
                  src={
                    this.state.selectedSong.source.length &&
                    this.state.selectedSong.source
                  }
                  onPlay={e => console.log("e2e2")}
                  onCanPlay={e => console.log(e)}
                  onEnded={e => {
                    if (!this.state.selectedSong.last) {
                      const song = this.props.songs.find(
                        item => item.id === this.state.selectedSong.id + 1
                      );
                      ponerCancion(song.audio, song.id);
                    }
                  }}
                />
              </Row>
            </Col>
            <Col>
              <Row style={{ "margin-left": "10%" }}>
                <Card style={{ width: "18rem" }}>
                  <ListGroup className="list-group-flush">
                    {this.props.songs.map(item => (
                      <ListGroupItem
                        onClick={() => ponerCancion(item.audio, item.id)}
                        style={{
                          cursor: "pointer",
                          background:
                            item.id === this.state.selectedSong.id
                              ? "aliceblue"
                              : "none"
                        }}
                      >
                        {
                          <Row>
                            <Col>{item.name}</Col>
                            <Col>
                              {new Date(item.seconds * 1000)
                                .toISOString()
                                .substr(14, 5)}
                              {" m"}
                            </Col>
                          </Row>
                        }
                      </ListGroupItem>
                    ))}
                  </ListGroup>
                </Card>
              </Row>
            </Col>
          </Row>
        </Container>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  fetchSongs: id => dispatch(fetchByAlbum(id))
});

export default connect(
  state => ({
    songs: getSongs(state),
    selectedAlbum: getSelectedAlbum(state)
  }),
  mapDispatchToProps
)(Album);
