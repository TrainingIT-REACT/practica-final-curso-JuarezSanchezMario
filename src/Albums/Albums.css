.Home {
  /*margin: 3em auto;
  max-width: 50em;*/
}

/*.App h1 {
  text-align: center;
}*/

.player rhap_button-clear rhap_main-controls-button rhap_play-pause-button{
     vertical-align: none;
}