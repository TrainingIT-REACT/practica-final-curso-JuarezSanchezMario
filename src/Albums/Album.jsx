import React, { Component } from "react";
import { connect } from "react-redux";
import { getSongs, getLoading } from "../reducers/songs";
import { fetchByAlbum, saveSongHistorico } from "../actions/songs";
import { getSelectedAlbum } from "../reducers/albums";
import Card from "react-bootstrap/Card";
import Container from "react-bootstrap/Container";
import ListGroup from "react-bootstrap/ListGroup";
import ListGroupItem from "react-bootstrap/ListGroupItem";
import AudioPlayer from "react-h5-audio-player";
import "react-h5-audio-player/lib/styles.css";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Spinner from "../utils/spinner.jsx";

// Css
import "./Albums.css";

class Album extends Component {
  constructor(props) {
    super(props);
    this.duration = 0;
    this.state = {
      selectedSong: {
        source: {},
        id: {},
        last: false
      },
    };
  }

  componentDidMount() {
    this.props.fetchSongs(this.props.selectedAlbum.id);
  }

  render() {
    if(this.props.loading){
      return <Spinner/>
    }
    const ponerCancion = item => {
      this.setState(
        {
          selectedSong: {
            source: ""
          }
        },
        () =>
          this.setState({
            selectedSong: {
              source: item.audio,
              id: item.id,
              last:
                item.id === this.props.songs[this.props.songs.length - 1].id
                  ? true
                  : false
            }
          })
      );
      this.props.saveSong(item);
    };
    if(!Object.keys(this.props.selectedAlbum).length){
      return("Ups no hay nada que mostrar");
    }
    return (
      <div className="Albums">
        <Container style={{ maxWidth: "100%", textAalign: "-webkit-center" }}>
          <Row>
            <Col lg="true">
              <Card style={{ width: "18rem" }}>
                <Card.Img alt="No image" variant="top" src={this.props.selectedAlbum.cover} />
                <Card.Body>
                  <Card.Title>{this.props.selectedAlbum.name}</Card.Title>
                  <Card.Text>{this.props.selectedAlbum.artist}</Card.Text>
                  <Card.Text>
                    {this.duration === 0
                      ? this.props.songs.forEach(
                          item => (this.duration += item.seconds)
                        )
                      : null}
                    {new Date(this.duration * 1000).toISOString().substr(14, 5)}
                    {" m"}
                  </Card.Text>
                </Card.Body>
              </Card>
              <Row>
                <AudioPlayer
                  className="player"
                  autoPlay={this.state.selectedSong.source.length}
                  src={
                    this.state.selectedSong.source.length &&
                    this.state.selectedSong.source
                  }
                  onPlay={e => console.log("e2e2")}
                  onCanPlay={e => console.log(e)}
                  onEnded={e => {
                    if (!this.state.selectedSong.last) {
                      const song = this.props.songs.find(
                        item => item.id === this.state.selectedSong.id + 1
                      );
                      ponerCancion(song);
                    }
                  }}
                />
              </Row>
            </Col>
            <Col lg="true">
              <Row style={{ marginLeft: "10%" }}>Pulsa para reproducir!</Row>
              <Row style={{ marginLeft: "10%" }}>
                <Card style={{ width: "18rem" }}>
                  <ListGroup className="list-group-flush">
                    {this.props.songs.map(item => (
                      <ListGroupItem
                        onClick={() => ponerCancion(item)}
                        style={{
                          cursor: "pointer",
                          background:
                            item.id === this.state.selectedSong.id
                              ? "aliceblue"
                              : "none"
                        }}
                      >
                        {
                          <Row>
                            <Col>{item.name}</Col>
                            <Col>
                              {new Date(item.seconds * 1000)
                                .toISOString()
                                .substr(14, 5)}
                              {" m"}
                            </Col>
                          </Row>
                        }
                      </ListGroupItem>
                    ))}
                  </ListGroup>
                </Card>
              </Row>
            </Col>
          </Row>
        </Container>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  fetchSongs: id => dispatch(fetchByAlbum(id)),
  saveSong: song => dispatch(saveSongHistorico(song))
});

export default connect(
  state => ({
    songs: getSongs(state),
    selectedAlbum: getSelectedAlbum(state),
    loading: getLoading(state)
  }),
  mapDispatchToProps
)(Album);
