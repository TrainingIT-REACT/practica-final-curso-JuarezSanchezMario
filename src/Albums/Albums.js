import React, { Component } from "react";
import { connect } from "react-redux";
import { getAlbums, getLoading } from "../reducers/albums";
import {
  savedSelectedAlbum,
  fetchFilteredByName,
  fetchFilteredByArtist
} from "../actions/albums";
import Card from "react-bootstrap/Card";
import CardColumns from "react-bootstrap/CardColumns";
import Spinner from "../utils/spinner.jsx";
import Badge from "react-bootstrap/Badge";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";

// Css
import "./Albums.css";

class Albums extends Component {
  constructor(props) {
    super(props);

    this.state = {
      lastSearch: "",
      artista:false,
      nombre:true,
    };
  }

  componentDidMount() {}

  handleSubmit(event, props) {
    const form = event.currentTarget;
    event.preventDefault();
    event.stopPropagation();
    if (form.formHorizontalRadios2.checked) {
      props.fetchByArits(form.search.value);
    } else {
      props.fetchByName(form.search.value);
    }
    this.setState({ lastSearch: form.search.value });
    console.log(event);
  }

  render() {
    if (this.props.loading) {
      return <Spinner />;
    }
    return (
      <div className="Albums">
        <Form onSubmit={event => this.handleSubmit(event, this.props)}>
          <Form.Group controlId="search">
            <Form.Control
              type="text"
              placeholder="Buscar"
              defaultValue={this.state.lastSearch}
            />
          </Form.Group>
          <Button variant="primary" type={"submit"}>
            Buscar
          </Button>
          <Form.Group>
            <Form.Check
              onClick={() => this.setState({nombre: true,artista:false})}
              type="radio"
              custom
              checked={this.state.nombre}
              label="Por nombre"
              name="formHorizontalRadios"
              id="formHorizontalRadios1"
            />
            <Form.Check
              type="radio"
              custom
              onClick={() => this.setState({artista: true,nombre:false})}
              checked={this.state.artista}
              label="Por artista"
              name="formHorizontalRadios"
              id="formHorizontalRadios2"
            />
          </Form.Group>
        </Form>
        <Badge variant="secondary">Albums</Badge>
        <CardColumns>
          {this.props.albums.map(item => (
            <Card
              style={{ cursor: "pointer" }}
              onClick={() => {
                this.props.dispatchSelectedAlbum(item);
                this.props.history.push("/album");
              }}
            >
              <Card.Img variant="top" src={item.cover} />
              <Card.Body>
                <Card.Title>{item.name}</Card.Title>
                <Card.Text>{item.artist}</Card.Text>
              </Card.Body>
            </Card>
          ))}
        </CardColumns>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  dispatchSelectedAlbum: album => dispatch(savedSelectedAlbum(album)),
  fetchByName: string => dispatch(fetchFilteredByName(string)),
  fetchByArits: string => dispatch(fetchFilteredByArtist(string))
});

export default connect(
  state => ({
    albums: getAlbums(state),
    loading: getLoading(state)
  }),
  mapDispatchToProps
)(Albums);
