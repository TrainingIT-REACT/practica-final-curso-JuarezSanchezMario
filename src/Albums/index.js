import Albums from './App';
export default Albums;
