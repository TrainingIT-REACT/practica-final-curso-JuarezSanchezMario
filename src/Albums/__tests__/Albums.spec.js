import React from 'react';
import { render } from 'enzyme';

// Componente
import Albums from '../Albums';

describe(Albums, () => {
  it('renders correctly', () => {
    const wrapper = render(<Albums />);
    expect(wrapper).toMatchSnapshot();
  });
})
