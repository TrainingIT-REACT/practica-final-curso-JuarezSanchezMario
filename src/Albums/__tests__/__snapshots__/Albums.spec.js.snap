// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums renders correctly 1`] = `
<div
  class="Albums"
>
  <header
    class="Home-header"
  >
    <p>
      Edit 
      <code>
        src/Albums.js
      </code>
       and save to reload.
    </p>
    <a
      class="Albums-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
